app/main/routes_frontend.py

```diff
@@ -171,14 +172,16 @@ def update_device_session(uid, session_type):
     update = request.get_json()
     if session_type == 'ferm':
         session = active_ferm_sessions[uid]
+    elif session_type == 'iSpindel':
+        session = active_iSpindel_sessions[uid]
 
         if update['active'] == False:
             session.active = False
             if session.file != None:
                 session.file.seek(0, os.SEEK_END)
                 if session.file.tell() > 0:
-                    # mark for completion and archive session file
-                    session.file.seek(session.file.tell() - 1, os.SEEK_SET)  # Remove trailing , from last data set
+                    # mark for completion and archive session file                   
+                    session.file.seek(session.file.tell() - 2, os.SEEK_SET)  # Remove trailing , and newline from last data set
                     session.file.write('\n]')
```

Comment:
can we instead just rely on the logic I added into `recover_active_session` from within `session_parser.py` this way all of this file seeking to close out an incomplete JSON happens in one location.

It is a bit strange we add a new line for ferm and ispindel (copied from ferm), but don't for pico, z or zymatic sessions... probably to keep consistent it might be a better idea to just remove trailing newline?

Reply:
I love that we string format and then send in a `json.dumps` 🤣